blackbox/handlers/storage/dropbox.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+import os
+from pathlib import Path
+
+from dropbox import Dropbox as DropboxClient
+from dropbox.exceptions import ApiError
+from dropbox.exceptions import HttpError
+from dropbox.files import CommitInfo
+from dropbox.files import FileMetadata
+from dropbox.files import UploadSessionCursor
+from dropbox.files import WriteMode
+
+from blackbox.config import Blackbox
+from blackbox.handlers.storage._base import BlackboxStorage
+from blackbox.utils.logger import log
+
+
+class Dropbox(BlackboxStorage):
+    """Storage handler that uploads backups to Dropbox."""
+
+    connstring_regex = r"dropbox://(?P<access_token>[^?]+)"
+    valid_prefixes = [
+        "dropbox"
+    ]
+
+    def __init__(self):
+        super().__init__()
+
+        self.success = False
+        self.output = ""
+
+        self.upload_base = self.config.get("upload_directory") or "/"
+        self.client = DropboxClient(self.config.get("access_token"))
+
+    def sync(self, file_path: Path) -> None:
+        """Sync a file to Dropbox."""
+        # This is size what can be uploaded as one chunk.
+        # When file is bigger than that, this will be uploaded
+        # in multiple parts.
+        chunk_size = 4 * 1024 * 1024
+
+        upload_path = f"{self.upload_base}{file_path.name}"
+
+        try:
+            with file_path.open("rb") as f:
+                file_size = os.path.getsize(str(file_path))
```

Comment:
optionally you can do this:
```python
file_size  = file_path.stat().st_size
```
then you don't need to import os just for that one call, but I'll leave it for you to decide

Reply:
Yeah, you should do this one. `os.path` is not necessary when we have `pathlib`